samtranslator/model/sam_resources.py

```diff
@@ -2540,6 +2539,100 @@ def _construct_ddb_datasource_connector_resources(
         )
         return connector.to_cloudformation(**kwargs)
 
+    def _construct_lambda_datasources(
+        self,
+        lambda_datasources: Optional[Dict[str, aws_serverless_graphqlapi.LambdaDataSource]],
+        api_id: Intrinsicable[str],
+        kwargs: Dict[str, Any],
+    ) -> List[Resource]:
+        if not lambda_datasources:
+            return []
+
+        resources: List[Resource] = []
+
+        for relative_id, lambda_datasource in lambda_datasources.items():
+            datasource_logical_id = f"{self.logical_id}Lambda{relative_id}"
+            cfn_datasource = DataSource(
+                logical_id=datasource_logical_id, depends_on=self.depends_on, attributes=self.resource_attributes
+            )
+
+            cfn_datasource.Name = lambda_datasource.Name or relative_id
+            cfn_datasource.Type = "AWS_LAMBDA"
+            cfn_datasource.ApiId = api_id
+            cfn_datasource.Description = passthrough_value(lambda_datasource.Description)
+            cfn_datasource.LambdaConfig = {"LambdaFunctionArn": passthrough_value(lambda_datasource.FunctionArn)}
+
+            cfn_datasource.ServiceRoleArn, permissions_resources = self._parse_lambda_datasource_role(
+                lambda_datasource,
+                cfn_datasource.get_runtime_attr("arn"),
+                lambda_datasource.FunctionArn,
+                datasource_logical_id,
+                kwargs,
+            )
+
+            self._datasource_name_map[relative_id] = cfn_datasource.get_runtime_attr("name")
+
+            resources.extend([cfn_datasource, *permissions_resources])
+
+        return resources
+
+    def _parse_lambda_datasource_role(
```

Comment:
Can also be static?

Reply:
It calls another function in the class on line 2601.

Reply:
That one's static.

Reply:
We also use it in the creation of `IAMRole`

Reply:
Ah right didn't notice :+1: